### PR TITLE
basic circleci support2

### PR DESCRIPTION
**Pull request by sagark**

### CODE REVIEW

**Comment by sagark** [comment]
this is ready for review

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine. See: https://circleci.com/docs/2.0/configuration-reference
+version: 2.1
+executors:
+    main-env:
+        docker:
+          - image: centos:centos7.6.1810
+        environment:
+            JVM_MEMORY: 2200m # Customize the JVM maximum heap limit
```
Comment (davidbiancolin) [inline .circleci/config.yml:8]: The JVM has a container mode that might play nicer than having to explicitly set something but if this works, it works.

File: .circleci/dockerprep.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# you will be root, so:
+adduser centos
+usermod -aG wheel centos
+
+# let the new acct run sudo without password
+echo "centos ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+
+# fixes to machine launch script to merge later:
+sudo yum -y install dtc # TODO: replace in machine-launch script, now available in epel
+# TODO: fix machine launch script, manually install the right numpy BEFORE matplotlib:
```
Comment (davidbiancolin) [inline .circleci/dockerprep.sh:24]: Can you open issues for this and the above?

**Comment by colinschmidt** [comment]
I'm excited to preserve all these `stuff` commits.

**Comment by davidbiancolin** [comment]
I read the commit messages and figured Sagar was writing lyrics for some trashy pop hook.

File: .circleci/dockerprep.sh (hunk 2 above)
Comment (sagark) [reply .circleci/dockerprep.sh:24]: done

File: .circleci/config.yml (hunk 1 above)
Comment (sagark) [reply .circleci/config.yml:8]: added to issues

**Pull request merged**